Mathlib/Analysis/NormedSpace/Unitization.lean

```diff
@@ -0,0 +1,243 @@
+        norm_add_le _ _
+      _ = ‖x.fst‖ + ‖x.snd‖ := by
+        rw [norm_algebraMap', (AddMonoidHomClass.isometry_iff_norm (mul 𝕜 A)).mp (isometry_mul 𝕜 A)]
+      _ ≤ _ := (add_le_add (le_max_left _ _) (le_max_right _ _)).trans_eq (two_mul _).symm
+
+open Bornology Filter
+open scoped Uniformity Topology
+
+theorem uniformity_eq_aux :
+    𝓤[instUniformSpaceProd.comap <| addEquiv 𝕜 A] = 𝓤 (Unitization 𝕜 A) := by
+  have key : UniformInducing (addEquiv 𝕜 A) :=
+    antilipschitzWith_addEquiv.uniformInducing lipschitzWith_addEquiv.uniformContinuous
+  rw [← key.comap_uniformity]
+  rfl
+
+theorem cobounded_eq_aux :
+    @cobounded _ (Bornology.induced <| addEquiv 𝕜 A) = cobounded (Unitization 𝕜 A) :=
+  le_antisymm lipschitzWith_addEquiv.comap_cobounded_le
+    antilipschitzWith_addEquiv.tendsto_cobounded.le_comap
+
+end Aux
+
+/-- The uniformity on `Unitization 𝕜 A` is inherited from `𝕜 × A`. -/
+instance instUniformSpace : UniformSpace (Unitization 𝕜 A) :=
+  instUniformSpaceProd.comap (addEquiv 𝕜 A)
+
+/-- The bornology on `Unitization 𝕜 A` is inherited from `𝕜 × A`. -/
+instance instBornology : Bornology (Unitization 𝕜 A) :=
+  Bornology.induced <| addEquiv 𝕜 A
+
+theorem uniformEmbedding_addEquiv : UniformEmbedding (addEquiv 𝕜 A) where
+  comap_uniformity := rfl
+  inj := (addEquiv 𝕜 A).injective
+
+/-- `Unitization 𝕜 A` is complete whenever `𝕜` and `A` are also.  -/
+instance instCompleteSpace [CompleteSpace 𝕜] [CompleteSpace A] :
+    CompleteSpace (Unitization 𝕜 A) :=
+  (completeSpace_congr uniformEmbedding_addEquiv).mpr CompleteSpace.prod
+
+/-- Pull back the metric structure from `𝕜 × (A →L[𝕜] A)` to `Unitization 𝕜 A` using the
+algebra homomorphism `Unitization.splitMul 𝕜 A`, but replace the bornology and the uniformity so
+that they coincide with `𝕜 × A`. -/
+noncomputable instance instMetricSpace : MetricSpace (Unitization 𝕜 A) :=
+  (normedRingAux.toMetricSpace.replaceUniformity uniformity_eq_aux).replaceBornology
+    fun s => Filter.ext_iff.1 cobounded_eq_aux (sᶜ)
+
+/-- Pull back the normed ring structure from `𝕜 × (A →L[𝕜] A)` to `Unitization 𝕜 A` using the
+algebra homomorphism `Unitization.splitMul 𝕜 A`. -/
+noncomputable instance instNormedRing : NormedRing (Unitization 𝕜 A)
+    where
+  dist_eq := normedRingAux.dist_eq
+  norm_mul := normedRingAux.norm_mul
+  norm := normedRingAux.norm
+
+/-- Pull back the normed algebra structure from `𝕜 × (A →L[𝕜] A)` to `Unitization 𝕜 A` using the
+algebra homomorphism `Unitization.splitMul 𝕜 A`. -/
+instance instNormedAlgebra : NormedAlgebra 𝕜 (Unitization 𝕜 A) where
+  norm_smul_le k x := by
+    rw [norm_def, map_smul, norm_smul, ← norm_def]
+
+instance instNormOneClass : NormOneClass (Unitization 𝕜 A) where
+  norm_one := by simpa only [norm_eq_sup, fst_one, norm_one, snd_one, map_one, map_zero,
+      add_zero, ge_iff_le, sup_eq_left] using op_norm_le_bound _ zero_le_one fun x => by simp
+
+lemma norm_inr (a : A) : ‖(a : Unitization 𝕜 A)‖ = ‖a‖ := by
```

Comment:
Nit: would be nice to have the nnnorm version too; and maybe dist and nudist lemmas

Reply:
I think I did what you wanted, but let me know if not.